users/models.js

```diff
@@ -33,12 +38,17 @@ UserSchema.methods.validatePassword = function(password) {
     .then(isValid => isValid);
 }
 
-UserSchema.statics.hashPassword = function(password) {
-  return bcrypt
-    .hash(password, 10)
-    .then(hash => hash);
-}
+UserSchema.pre('save', function(next) {
+    var user = this;
+    // only hash the password if it has been modified (or is new)
+    if (!user.isModified('password')) return next();
+    bcrypt.hash(user.password, 10, function(err, hash) {
+      if (err) return next(err);
+      user.password = hash;
```

Comment:
Is the idea that `user` sticks around? Trying to grok who cares about `user.password` since we don't return it or do anything with it.